pkg/apis/batch/validation/validation.go

```diff
@@ -425,6 +548,26 @@ func validateJobStatus(status *batch.JobStatus, fldPath *field.Path) field.Error
 			}
 		}
 	}
+	if opts.RejectSuccessCriteriaMetForNonIndexedJob {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("conditions"), field.OmitValueType{}, "cannot set SuccessCriteriaMet to NonIndexed Job"))
+		}
+	}
+	if opts.RejectSuccessCriteriaMetWithFailedCondition {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) && IsJobFailed(job) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("conditions"), field.OmitValueType{}, "cannot set SuccessCriteriaMet=True and Failed=true conditions"))
+		}
+	}
+	if opts.RejectSuccessCriteriaMetWithFailureTargetCondition {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) && IsConditionTrue(job.Status.Conditions, batch.JobFailureTarget) {
+			allErrs = append(allErrs, field.Invalid(fldPath.Child("conditions"), field.OmitValueType{}, "cannot set SuccessCriteriaMet=True and FailureTarget=true conditions"))
+		}
+	}
+	if opts.RejectSuccessCriteriaMetForAlreadyCompleteJob {
+		if IsConditionTrue(job.Status.Conditions, batch.JobSuccessCriteriaMet) {
```

Comment:
I'm trying to block the adding of SuccessCriteriaMet condition after the Job gets `Complete` condition.
Also, we already verified the other transitions in the `RejectSuccessCriteriaMetWithFailureTargetCondition` and `RejectSuccessCriteriaMetWithFailedCondition`. In case of`Completion`, we allow jobs the transition `SuccessCriteriaMet` -> `Complete`.

-  disallowed
    - FailureTarget > SuccessCriteriaMet
    - SuccessCriteriaMet > FailureTarget
    - Failed -> SuccessCriteriaMet
    - SuccessCriteriaMet -> Failure
    - Complete -> SuccessCriteriaMet (Here is validation for this)
- allowed
    - SuccessCriteriaMet -> Complete